pkg/route/validation/validation_test.go

```diff
@@ -833,6 +835,355 @@ func TestValidateRoute(t *testing.T) {
+					To:             createRouteSpecTo("serviceName", "Service"),
+					WildcardPolicy: routev1.WildcardPolicySubdomain,
+					HTTPHeaders: routev1.RouteHTTPHeaders{
+
+						Actions: routev1.RouteHTTPHeaderActions{
+							Response: []routev1.RouteHTTPHeader{
+								{
+									Name: "map",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "%[path,map_beg(/etc/hapee-2.6/routes.map,webservers)]",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrors: 1,
+		},
+		{
+			name: "Valid headers",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcardpolicy",
+					Namespace: "foo",
+				},
+				Spec: routev1.RouteSpec{
+					Host:           "subdomain.wildcard.test",
+					To:             createRouteSpecTo("serviceName", "Service"),
+					WildcardPolicy: routev1.WildcardPolicySubdomain,
+					HTTPHeaders: routev1.RouteHTTPHeaders{
+
+						Actions: routev1.RouteHTTPHeaderActions{
+							Response: []routev1.RouteHTTPHeader{
+								{
+									Name: headerNameXFrame,
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "DENY",
+										},
+									},
+								},
+								{
+									Name: headerNameXSS,
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "1;mode=block",
+										},
+									},
+								},
+								{
+
+									Name: headerNameXFrame,
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Delete,
+									},
+								},
+								{
+
+									Name: headerNameXSS,
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Delete,
+									},
+								},
+							},
+							Request: []routev1.RouteHTTPHeader{
+								{
+									Name: "Accept",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "text/plain,text/html",
+										},
+									},
+								},
+								{
+
+									Name: "Accept-Encoding",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Delete,
+									},
+								},
+								{
+									Name: "Condtional",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "%[req.hdr(Host)] if foo",
+										},
+									},
+								},
+								{
+									Name: "Condition",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: `%[req.hdr(Host)]\ if\ foo`,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrors: 0,
```

Comment:
It would also be useful to check the headers at the end to see whether the headers actions are performed in the order given.  Does it always delete after it sets?  This is may be a validation test needed in another place.

Reply:
delete happens for the existing header however what value the haproxy.template had can't be fetched here.